pkg/cmdline/cmd.go

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package cmdline
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+// CommandManager holds root command or first parent
+// and can stores group of command
+type CommandManager struct {
+	rootCmd   *cobra.Command
+	groupCmds map[string][]*cobra.Command
+	errPool   []error
+	fm        *flagManager
+}
+
+// FlagError represents an error type returnes when command
+// line parsing fails
+type FlagError string
+
+func (f FlagError) Error() string {
+	return string(f)
+}
+
+func onError(cmd *cobra.Command, err error) error {
+	return FlagError(err.Error())
+}
+
+// NewCommandManager instantiates a CommandManager
+func NewCommandManager(rootCmd *cobra.Command) *CommandManager {
+	if rootCmd == nil {
+		panic("nil root command passed")
+	}
+	cm := &CommandManager{
+		rootCmd:   rootCmd,
+		groupCmds: make(map[string][]*cobra.Command),
+		fm:        newFlagManager(),
+	}
+	rootCmd.SetFlagErrorFunc(onError)
+	return cm
+}
+
+func (m *CommandManager) isRegistered(cmd *cobra.Command) bool {
+	c := cmd.Parent()
+	for c != nil {
+		if c == m.rootCmd {
+			break
+		}
+		c = c.Parent()
+	}
+	return c != nil
+}
+
+func (m *CommandManager) pushError(err error) {
+	m.errPool = append(m.errPool, err)
+}
+
+// GetError returns the error pool
+func (m *CommandManager) GetError() []error {
+	return m.errPool
+}
+
+// RegisterCmd registers a child command for the root command
+func (m *CommandManager) RegisterCmd(cmd *cobra.Command, interspersed bool) {
+	// panic here because it's a misuse of API and generally from
+	// global context or init() functions
+	if cmd == nil {
+		panic("nil command passed")
+	}
+	cmd.SetFlagErrorFunc(onError)
+	m.rootCmd.AddCommand(cmd)
+	cmd.Flags().SetInterspersed(interspersed)
+	m.SetCmdGroup(m.GetCmdName(cmd), cmd)
+}
+
+// RegisterSubCmd registers a child command for parent command given as argument
+func (m *CommandManager) RegisterSubCmd(parentCmd, childCmd *cobra.Command, interspersed bool) {
+	// panic here because it's a misuse of API and generally from
+	// global context or init() functions
+	if parentCmd == nil {
+		panic("nil parent command passed")
+	} else if childCmd == nil {
+		panic("nil child command passed")
+	} else if !m.isRegistered(parentCmd) {
+		panic("parent command not registered")
```

Comment:
Those checks are here to catch potential dev mistakes, nothing else.

Yes will remove interspersed argument, it could be set outside of CLI manager if needed